src/DeviceInput/InputDevices/internalDeviceSourceManager.ts

```diff
@@ -115,9 +110,33 @@ export class InternalDeviceSourceManager implements IDisposable {
         return deviceArray;
     }
 
-    public dispose(): void {
-        this.onDeviceConnectedObservable.clear();
-        this.onDeviceDisconnectedObservable.clear();
+    public registerManager = (manager: IObservableManager): void => {
+        this._registeredManagers.push(manager);
+    }
+
+    public unregisterManager = (manager: IObservableManager): void => {
+        const idx = this._registeredManagers.indexOf(manager);
+
+        if (idx > -1) {
+            this._registeredManagers.splice(idx, 1);
+        }
+    }
+
+    public enableEvents = (): void => {
+        this._deviceInputSystem.configureEvents();
+    }
+
+    public disableEvents = (): void => {
+        this._deviceInputSystem.unconfigureEvents();
+    }
+
+    public dispose = (): void => {
+        // If we have managers that weren't properly disposed of, dispose of their observables.

```

Comment:
I don't think this is necessary. The DSM's dispose will do this already.

Reply:
Removed and set up DSMs to handle their own dispose calls